lib/charms/opensearch/v0/opensearch_snapshots.py

```diff
@@ -0,0 +1,1197 @@
+        )
+        logger.debug("Snapshot repository creation response: %s", response)
+
+        # This should always pass and is set for documentation purposes
+        assert response.get("acknowledged") is True
+        return repo_name
+
+    @retry(stop=stop_after_attempt(3), wait=wait_fixed(3), reraise=True)
+    def remove_repo(self, object_storage_type: ObjectStorageType) -> None:
+        """Remove the current repository."""
+        repo_name = self.repository_name(object_storage_type)
+        try:
+            response = self.opensearch.request(
+                "DELETE", f"_snapshot/{repo_name}", alt_hosts=self.charm.alt_hosts
+            )
+            logger.debug("Snapshot repository creation response: %s", response)
+
+            # This should always pass and is set for documentation purposes
+            assert response.get("acknowledged") is True
+        except OpenSearchHttpError as e:
+            # we might be attempting to delete a nonexisting repository
+            if e.response_body.get("error", {}).get("type") == "repository_missing_exception":
+                return
+            raise
+
+    @retry(stop=stop_after_attempt(3), wait=wait_fixed(3), reraise=True)
+    def create_snapshot(self, object_storage_type: ObjectStorageType) -> str:
+        """Create an OpenSearch snapshot."""
+        repo_name = self.repository_name(object_storage_type)
+        snapshot_id = datetime.now().strftime(OPENSEARCH_BACKUP_ID_FORMAT).lower()
+        ignore = [f"-{idx}" for idx in SYSTEM_INDICES]
+        indices_clause = ",".join(["*"] + ignore)
+        logger.info("indices_clause: %s", indices_clause)
+        # create snapshot
+        response = self.opensearch.request(
+            "PUT",
+            f"_snapshot/{repo_name}/{snapshot_id}?wait_for_completion=false",
+            payload={"indices": indices_clause, "ignore_unavailable": True},
+            alt_hosts=self.charm.alt_hosts,
+            timeout=30,
+        )
+
+        logger.info(f"Snapshot request submitted with backup-id: {snapshot_id}")
+        logger.debug(f"Create snapshot request with id: {snapshot_id} - response: {response}")
+
+        # This should always pass and is set for documentation purposes
+        assert response.get("accepted") is True
+
+        return snapshot_id
+
+    @retry(
+        stop=stop_after_attempt(3),
+        wait=wait_fixed(3),
+        retry=retry_if_exception_type(OpenSearchHttpError),
+        reraise=True,
+    )
+    def restore_snapshot(
+        self, object_storage_type: ObjectStorageType, snapshot: dict[str, Any]
+    ) -> set[str]:
+        """Restore an OpenSearch snapshot."""
+        repo_name = self.repository_name(object_storage_type)
+        snapshot_id = snapshot.get("snapshot")
+
+        # This is not necessary if our charm performed the backup itself, but we add this
+        # as a safeguard against manually run backups (through direct calls to the rest api)
+        indices_to_ignore = "-,".join(
```

Comment:
This is currently producing `'.charm_node_lock-,.opendistro_security'` which is not going to ignore the indices

```suggestion
        indices_to_ignore = ",-".join(SYSTEM_INDICES)
```